pkg/daemon/ceph/osd/remove.go

```diff
@@ -140,6 +140,13 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 	// call archiveCrash to silence crash warning in ceph health if any
 	archiveCrash(clusterdContext, clusterInfo, osdID)
 
+	// remove the auth for the osd. should be done by 'osd purge', but doesn't always seem to happen
+	err = client.AuthDelete(clusterdContext, clusterInfo, fmt.Sprintf("osd.%d", osdID))
```

Comment:
Are we sure that the osd purge command isn't actually removing the auth? I haven't been clear in those reports if the auth was actually left behind after the purge, or if the creation of a new OSD was just confused when it tried to use the same OSD ID, or the same disk.

Reply:
I feel like it does not harm but at the same time finding the root cause would be good too. If the `purge` command does not remove the cephx (entirely possible), then we would certainly have a trace of that failure somewhere.

But for the time-being we can go ahead with that.

Reply:
I think this helps us find the root cause also. If we continue to get issues about OSDs not deploying because auth exists, then I think we can be more sure that it's *not* about the purge job failing to remove the auth but because the creation of the new OSD gets confused.

Reply:
Agreed it doesn't do harm and may help troubleshoot, would just be good to understand the root cause. No problem to merge for now.